src/toil/utils/toilKill.py

```diff
@@ -25,28 +26,47 @@
 
 def main() -> None:
     parser = parser_with_common_options()
+    parser.add_argument('--force', action='store_true',
+                        help="Send SIGKILL to the leader process if local.")
     options = parser.parse_args()
     set_logging_from_options(options)
     config = Config()
     config.setOptions(options)
 
-    # Kill the pid recorded in the job store.
+    # Get the job store
     try:
         job_store = Toil.resumeJobStore(config.jobStore)
     except NoSuchJobStoreException:
         logger.error("The job store %s does not exist.", config.jobStore)
         return
 
+    # Get the leader PID
     with job_store.read_shared_file_stream("pid.log") as f:
         pid_to_kill = int(f.read().strip())
 
-    # TODO: We assume this is a local PID so we can't kill the leader if the
-    #  process is on a different machine. We also rely on the leader to be
-    #  responsive to clean up all its jobs.
+    # Check if the leader is on the same machine
+    with job_store.read_shared_file_stream("leader_node_id.log") as f:
+        leader_node_id = f.read().decode('utf-8').strip()
```

Comment:
Could this be a reciprocal `read_node_id_file()` function alongside `write_node_id_file()` above?

Reply:
Maybe. It would be weird for it to be a member of `toil.common.Toil` though. Maybe just in this file?

Reply:
That makes sense to me.